### ISSUE TITLE
Replicate log based metrics in open census.

### ISSUE THREAD

**Issue by icco** [google/exposure-notifications-server#977]
### TL;DR

We currently have a bunch of metrics that are exported as log metrics, lets move them to use the pkg/observability code.

### Design

**Proposal**

Where ever we are calling metrics.Write, also send to open census. Example places:

 - https://github.com/google/exposure-notifications-server/blob/702c0796ecf6ab1a6816aaa2dfe422fed39fce9d/internal/export/batcher.go#L86
 - https://github.com/google/exposure-notifications-server/blob/64e2c6eb7bdad1178582512b2fab79f4d7f6e72b/internal/publish/publish.go#L197
 - https://github.com/google/exposure-notifications-server/blob/64e2c6eb7bdad1178582512b2fab79f4d7f6e72b/internal/publish/publish.go#L215

Also metric names in open census should have `_` not `-`.

**Alternatives considered**

Ignore. Probably fine, but extra visibility is always good.

**Resources**

- An example of some metrics in the verification server: https://github.com/google/exposure-notifications-verification-server/blob/2cda89fcf1a0f963bf57d06e97346b9fe4c594cc/pkg/controller/issueapi/metrics.go
- godoc: https://pkg.go.dev/go.opencensus.io/stats 

**Additional information**
